fil_fungible_token/src/token/mod.rs

```diff
@@ -115,22 +115,81 @@ where
     /// it doesn't exist
     ///
     /// If the account cannot be created, this function returns MessagingError::AddressNotInitialized
-    fn resolve_to_id(&self, address: &Address) -> MessagingResult<ActorID> {
+    fn resolve_or_init(&self, address: &Address) -> MessagingResult<ActorID> {
         let id = match self.msg.resolve_id(address) {
             Ok(addr) => addr,
-            Err(MessagingError::AddressNotInitialized(_e)) => {
-                self.msg.initialize_account(address)?
-            }
+            Err(MessagingError::AddressNotResolved(_e)) => self.msg.initialize_account(address)?,
             Err(e) => return Err(e),
         };
         Ok(id)
     }
 
-    /// Attempts to resolve an address to an ID address, returning MessagingError::AddressNotInitialized
+    /// Attempts to resolve an address to an ActorID, returning MessagingError::AddressNotResolved
     /// if it wasn't found
     fn get_id(&self, address: &Address) -> MessagingResult<ActorID> {
         self.msg.resolve_id(address)
     }
+
+    /// Attempts to compare two addresses, seeing if they would resolve to the same Actor without
```

Comment:
Style Nit: Consider moving helpers toward the bottom of the file (a new impl block?) so that the code reads more in a top-down fashion, when read in order.